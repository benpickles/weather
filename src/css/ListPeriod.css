.ListPeriod {
  border-top: 1px solid #ccc;
  cursor: pointer;
  padding: 5px;
}

.ListPeriod:first-child {
  border-top: none;
}

.ListPeriod:hover {
  background: #ccc;
}
