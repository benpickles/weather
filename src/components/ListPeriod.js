import React from 'react'
import moment from 'moment'
import '../css/ListPeriod.css'

export default ({ period }) =>
  <div className="ListPeriod">
    {moment.unix(period.dt).format('HH:mm')}
  </div>
