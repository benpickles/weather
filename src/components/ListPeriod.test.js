import React from 'react'
import renderer from 'react-test-renderer'
import ListPeriod from './ListPeriod'

it('renders correctly', () => {
  const period = { dt: 1493215200 }
  let selected

  const tree = renderer.create(
    <ListPeriod
      onSelectPeriod={p => selected = p}
      period={period}
    />
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

it('yields the clicked period', () => {
  const period = { dt: 1493215200 }
  let selected

  const component = renderer.create(
    <ListPeriod
      onSelectPeriod={p => selected = p}
      period={period}
    />
  )

  const tree = component.toJSON()
  tree.props.onClick()

  expect(selected).toEqual(period)
})
