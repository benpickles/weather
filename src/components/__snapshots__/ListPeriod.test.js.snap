exports[`test renders correctly 1`] = `
<div
  className="ListPeriod"
  onClick={[Function]}>
  15:00
</div>
`;
